tempesta_fw/cache.c

```diff
@@ -573,6 +626,8 @@ tfw_cache_add(TfwHttpResp *resp, TfwHttpReq *req)
 
 	if (!cache_cfg.cache || !tfw_cache_msg_cacheable(req))
 		return true;
+	if (!tfw_cache_employ(req, resp))
```

Comment:
Ok, agree. However, the main point was that you do extra check for NULL, but you have the knowledge about ingress arguments in most of the cases.